watertap/core/tests/test_zero_order_unit.py

```diff
@@ -0,0 +1,677 @@
+        assert len(model.fs.unit.solute_outlet_equation) == 3
+        assert isinstance(model.fs.unit.outlet_pressure_constraint, Constraint)
+        assert len(model.fs.unit.outlet_pressure_constraint) == 1
+        assert isinstance(model.fs.unit.waste_pressure_constraint, Constraint)
+        assert len(model.fs.unit.waste_pressure_constraint) == 1
+        assert isinstance(model.fs.unit.outlet_temperature_equality,
+                          Constraint)
+        assert len(model.fs.unit.outlet_temperature_equality) == 1
+        assert isinstance(model.fs.unit.waste_temperature_equality, Constraint)
+        assert len(model.fs.unit.waste_temperature_equality) == 1
+
+    @pytest.mark.unit
+    def test_degrees_of_freedom(self, model):
+        assert degrees_of_freedom(model) == 0
+
+    @pytest.mark.component
+    def test_unit_consistency(self, model):
+        assert_units_consistent(model)
+
+    @pytest.mark.component
+    def test_scaling(self, model):
+        iscale.calculate_scaling_factors(model)
+
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.water_recovery_equation[0]) == 1e3
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.flow_balance[0]) == 1e3
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.solute_removal_equation[0, "A"]) == 1e5
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.solute_removal_equation[0, "B"]) == 1e5
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.solute_removal_equation[0, "C"]) == 1e5
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.solute_outlet_equation[0, "A"]) == 1e5
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.solute_outlet_equation[0, "B"]) == 1e5
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.solute_outlet_equation[0, "C"]) == 1e5
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.outlet_pressure_constraint[0]) == 1e-5
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.waste_pressure_constraint[0]) == 1e-5
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.outlet_temperature_equality[0]) == 1e-2
+        assert iscale.get_constraint_transform_applied_scaling_factor(
+            model.fs.unit.outlet_temperature_equality[0]) == 1e-2
+
+    @pytest.mark.component
+    def test_initialization(self, model):
+        initialization_tester(model)
+
+    @pytest.mark.component
+    def test_solve(self, model):
+        results = solver.solve(model)
+
+        # Check for optimal solution
+        assert results.solver.termination_condition == \
```

Comment:
Same as above (although this works and no dire need to change if you don't want to)